src/data/real/ennreal.lean

```diff
@@ -1268,6 +1273,107 @@ begin
+               inv_pos, false_and] }
+end
+
+lemma zpow_lt_top (ha : a ≠ 0) (h'a : a ≠ ∞) (n : ℤ) : a ^ n < ∞ :=
+begin
+  cases n,
+  { exact ennreal.pow_lt_top h'a.lt_top _ },
+  { simp only [ennreal.pow_pos ha.bot_lt (n + 1), zpow_neg_succ_of_nat, inv_lt_top] }
+end
+
+lemma exists_mem_Ico_zpow
+  {x y : ℝ≥0∞} (hx : x ≠ 0) (h'x : x ≠ ∞) (hy : 1 < y) (h'y : y ≠ ⊤) :
+  ∃ n : ℤ, x ∈ Ico (y ^ n) (y ^ (n + 1)) :=
+begin
+  lift x to ℝ≥0 using h'x,
+  lift y to ℝ≥0 using h'y,
+  have A : y ≠ 0, by simpa only [ne.def, coe_eq_zero] using (ennreal.zero_lt_one.trans hy).ne',
+  obtain ⟨n, hn, h'n⟩ : ∃ n : ℤ, y ^ n ≤ x ∧ x < y ^ (n + 1),
+  { refine nnreal.exists_mem_Ico_zpow _  (one_lt_coe_iff.1 hy),
+    simpa only [ne.def, coe_eq_zero] using hx },
+  refine ⟨n, _, _⟩,
+  { rwa [← ennreal.coe_zpow A, ennreal.coe_le_coe] },
+  { rwa [← ennreal.coe_zpow A, ennreal.coe_lt_coe] }
+end
+
+lemma exists_mem_Ioc_zpow
+  {x y : ℝ≥0∞} (hx : x ≠ 0) (h'x : x ≠ ∞) (hy : 1 < y) (h'y : y ≠ ⊤) :
+  ∃ n : ℤ, x ∈ Ioc (y ^ n) (y ^ (n + 1)) :=
+begin
+  lift x to ℝ≥0 using h'x,
+  lift y to ℝ≥0 using h'y,
+  have A : y ≠ 0, by simpa only [ne.def, coe_eq_zero] using (ennreal.zero_lt_one.trans hy).ne',
+  obtain ⟨n, hn, h'n⟩ : ∃ n : ℤ, y ^ n < x ∧ x ≤ y ^ (n + 1),
+  { refine nnreal.exists_mem_Ioc_zpow _  (one_lt_coe_iff.1 hy),
+    simpa only [ne.def, coe_eq_zero] using hx },
+  refine ⟨n, _, _⟩,
+  { rwa [← ennreal.coe_zpow A, ennreal.coe_lt_coe] },
+  { rwa [← ennreal.coe_zpow A, ennreal.coe_le_coe] }
+end
+
+lemma Ioo_zero_top_eq_Union_Ico_zpow {y : ℝ≥0∞} (hy : 1 < y) (h'y : y ≠ ⊤) :
+  Ioo (0 : ℝ≥0∞) (∞ : ℝ≥0∞) = ⋃ (n : ℤ), Ico (y^n) (y^(n+1)) :=
+begin
+  ext x,
+  simp only [mem_Union, mem_Ioo, mem_Ico],
+  split,
+  { rintros ⟨hx, h'x⟩,
+    exact exists_mem_Ico_zpow hx.ne' h'x.ne hy h'y },
+  { rintros ⟨n, hn, h'n⟩,
+    split,
+    { apply lt_of_lt_of_le _ hn,
+      exact ennreal.zpow_pos (ennreal.zero_lt_one.trans hy).ne' h'y _ },
+    { apply lt_trans h'n _,
+      exact ennreal.zpow_lt_top (ennreal.zero_lt_one.trans hy).ne' h'y _ } }
+end
+
+lemma zpow_le_of_le {x : ℝ≥0∞} (hx : 1 ≤ x) {a b : ℤ} (h : a ≤ b) : x ^ a ≤ x ^ b :=
+begin
+  induction a with a a; induction b with b b,
+  { simp,
+    apply pow_le_pow hx,
+    apply int.le_of_coe_nat_le_coe_nat h },
+  { apply absurd h,
+    apply not_le_of_gt,
+    exact lt_of_lt_of_le (int.neg_succ_lt_zero _) (int.of_nat_nonneg _) },
+  { simp only [zpow_neg_succ_of_nat, int.of_nat_eq_coe, zpow_coe_nat],
+    refine le_trans (inv_le_one.2 _) _;
+    apply ennreal.one_le_pow_of_one_le hx, },
+  { simp only [zpow_neg_succ_of_nat],
+    apply inv_le_inv.2,
+    { apply pow_le_pow hx,
+      have : -(↑(a+1) : ℤ) ≤ -(↑(b+1) : ℤ), from h,
+      have h' := le_of_neg_le_neg this,
+      apply int.le_of_coe_nat_le_coe_nat h' },
+    repeat { apply pow_pos (lt_of_lt_of_le zero_lt_one hx) } }
+end
+
+lemma monotone_zpow {x : ℝ≥0∞} (hx : 1 ≤ x) : monotone (λ (a : ℤ), x ^ a) :=
```

Comment:
This expands better without the binder,
```suggestion
lemma monotone_zpow {x : ℝ≥0∞} (hx : 1 ≤ x) : monotone ((^) x : ℤ → ℝ≥0∞) :=
```
Otherwise `monotone_zpow hx hab` expands to something that's hard to rewrite by